src/main/java/com/softserve/repository/SortOrderRepository.java

```diff
@@ -0,0 +1,113 @@
+package com.softserve.repository;
+
+import com.softserve.entity.interfaces.SortableOrder;
+import com.softserve.exception.EntityNotFoundException;
+import com.softserve.exception.SortOrderNotExistsException;
+import lombok.extern.slf4j.Slf4j;
+import org.hibernate.SessionFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.ConfigurableBeanFactory;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.TypedQuery;
+import java.util.Optional;
+
+@Slf4j
+@Repository
+@Scope(value = ConfigurableBeanFactory.SCOPE_PROTOTYPE)
+public class SortOrderRepository<T extends SortableOrder> {
+    public SortOrderRepository() {
+    }
+
+    private Class<?> tClass = SortOrderRepository.class;
```

Comment:
the setter is a bid bizarre in this case, let's follow this naming convention:
https://stackoverflow.com/questions/2529974/why-do-java-programmers-like-to-name-a-variable-clazz